src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,243 @@
+            return deleteCarByIdStatement.executeUpdate() > 0;
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't delete car by id: " + id,
+                    throwables);
+        }
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        String addDriverQuery = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addDriverStatement = connection
+                        .prepareStatement(addDriverQuery)) {
+            addDriverStatement.setLong(1, driver.getId());
+            addDriverStatement.setLong(2, car.getId());
+            addDriverStatement.executeUpdate();
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't add driver: "
+                    + driver + " to car: "
+                    + car, throwables);
+        }
+
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        String removeDriverFromCarQuery = "DELETE FROM cars_drivers "
+                + "WHERE driver_id = ? AND car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteDriverFromCarStatement = connection
+                        .prepareStatement(removeDriverFromCarQuery)) {
+            deleteDriverFromCarStatement.setLong(1, driver.getId());
+            deleteDriverFromCarStatement.setLong(2, car.getId());
+            deleteDriverFromCarStatement.executeUpdate();
+
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't remove driver: "
+                    + driver + " from car: " + car, throwables);
+        }
+
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsByDriverIdQuery = "SELECT c.id , c.model, c.manufacturer_id,"
+                + " m.name, m.country "
+                + "FROM cars_drivers cd "
+                + "JOIN cars c "
+                + "ON cd.car_id = c.id "
+                + "JOIN drivers d "
+                + "ON cd.driver_id = d.id "
+                + "JOIN manufacturers m "
+                + "ON m.id = c.manufacturer_id "
+                + "WHERE c.is_deleted = false AND d.is_deleted = false AND cd.driver_id = ?;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsByDriverIdIdStatement = connection
+                        .prepareStatement(getAllCarsByDriverIdQuery)) {
+            getCarsByDriverIdIdStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsByDriverIdIdStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarFromResultSet(resultSet);
+                car.setDrivers(getDriversByCar(car));
```

Comment:
you open connection to db inside of another connection, that's a bad practice. Init drivers after closing try block